node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -46,6 +47,8 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
 
   val peer: Peer
 
+  val initialSyncDone: Option[Promise[Done]]
```

Comment:
I don't understand what is happening here. Can you add documentation to it? It's also concerning that this is public and can be completed completed by anyone. Shouldn't `DataMessageHandler` just have the reference to the `Promise` and everyone else that is using this would just have a `initSyncDoneF: Option[Future[Done]]`?

Reply:
This is an initial sync latch,  eg:
https://github.com/rorp/eclair/blob/bitcoin-s-integration/eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala#L148